src/libexpr/primops/flake.hh

```diff
@@ -36,16 +36,16 @@ struct LockFile
     };
 
     std::map<FlakeRef, FlakeEntry> flakeEntries;
-    std::map<FlakeId, NonFlakeEntry> nonFlakeEntries;
+    std::map<FlakeAlias, NonFlakeEntry> nonFlakeEntries;
 };
 
 typedef std::vector<std::shared_ptr<FlakeRegistry>> Registries;
 
 Path getUserRegistryPath();
 
-enum RegistryAccess { DisallowRegistry, AllowRegistry, AllowRegistryAtTop };
+enum HandleLockFile { AllPure, UpdateLockFile, UseUpdatedLockFile, RecreateLockFile, UseNewLockFile};
```

Comment:
Could you add some comments explaining the semantics of each mode? It's not really clear from the name what the difference between `UpdateLockFile` and `UseUpdateLockFile` is, or between `RecreateLockFile` and `UseNewLockFile`.